Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import OpenTelemetryApi
+import OpenTelemetrySdk
+
+struct ZipkinConversionExtension {
+    static let statusCode = "ot.status_code"
+    static let statusDescription = "ot.status_description"
+
+    static let remoteEndpointServiceNameKeyResolution = ["peer.service": 0,
+                                                         "net.peer.name": 1,
+                                                         "peer.hostname": 2,
+                                                         "peer.address": 2,
+                                                         "http.host": 3,
+                                                         "db.instance": 4]
+
+    static var localEndpointCache = [String: ZipkinEndpoint]()
+    static var remoteEndpointCache = [String: ZipkinEndpoint]()
+
+    struct AttributeEnumerationState {
+        var tags = [String: String]()
+        var RemoteEndpointServiceName: String?
+        var remoteEndpointServiceNamePriority: Int?
+        var serviceName: String?
+        var serviceNamespace: String?
+    }
+
+    static func toZipkinSpan(otelSpan: SpanData, defaultLocalEndpoint: ZipkinEndpoint, useShortTraceIds: Bool = false) -> ZipkinSpan {
+        let parentId = otelSpan.parentSpanId.hexString
+
+        var attributeEnumerationState = AttributeEnumerationState()
+
+        otelSpan.attributes.forEach {
+            processAttributes(state: &attributeEnumerationState, key: $0.key, value: $0.value)
+        }
+
+        otelSpan.resource.attributes.forEach {
+            processResources(state: &attributeEnumerationState, key: $0.key, value: $0.value)
+        }
+
+        var localEndpoint = defaultLocalEndpoint
+
+        var serviceName = attributeEnumerationState.serviceName
+
+        if serviceName != nil, !serviceName!.isEmpty {
+            if let serviceNamespace = attributeEnumerationState.serviceNamespace, !serviceNamespace.isEmpty {
+                serviceName = serviceNamespace + "." + serviceName!
```

Comment:
I think this logic was implemented in places, but spec forbid to combine namespace and name

Reply:
True, fixed and added the namespace as a Zipkin tag as the spec says.